contracts/managers/allocators/SocialAllocator.sol

```diff
@@ -0,0 +1,401 @@
+     * @param  _pricePrecision              Amount of significant figures kept in determining new units
+     */
+    constructor(
+        ERC20Detailed _baseAsset,
+        ERC20Detailed _quoteAsset,
+        IOracle _baseAssetOracle,
+        IOracle _quoteAssetOracle,
+        ICore _core,
+        address _setTokenFactory,
+        uint256 _pricePrecision
+    )
+        public
+    {
+        baseAsset = _baseAsset;
+        quoteAsset = _quoteAsset;
+
+        baseAssetOracle = _baseAssetOracle;
+        quoteAssetOracle = _quoteAssetOracle;
+
+        // Query decimals of base and quote assets
+        baseAssetDecimals = _baseAsset.decimals();
+        quoteAssetDecimals = _quoteAsset.decimals();
+
+        core = _core;
+        setTokenFactory = _setTokenFactory;
+        pricePrecision = _pricePrecision;
+
+        // Calculate constants that will be used in calculations
+        uint256 minDecimals = Math.min(baseAssetDecimals, quoteAssetDecimals);
+
+        // Decimal difference for asset a1 is 10 ** (a1Decimals - min(a1Decimals, a2Decimals))
+        baseAssetDecimalDifference = CommonMath.safePower(10, uint256(baseAssetDecimals).sub(minDecimals));
+        quoteAssetDecimalDifference = CommonMath.safePower(10, uint256(quoteAssetDecimals).sub(minDecimals));
+
+        // NaturalUnit is equal to max(a1DecimalDifference, a2DecimalDifference) * pricePrecision
+        collateralNaturalUnit = Math.max(baseAssetDecimalDifference, quoteAssetDecimalDifference)
+            .mul(pricePrecision);
+
+        // Next set components will always be in order of base asset first
+        multiAssetNextSetComponents = [address(baseAsset), address(quoteAsset)];
+        baseAssetNextSetComponents = [address(baseAsset)];
+        quoteAssetNextSetComponents = [address(quoteAsset)];
+    }
+
+    /* ============ External ============ */
+
+    /*
+     * Determine the next allocation to rebalance into.
+     *
+     * @param  _targetBaseAssetAllocation       Target allocation of the base asset
+     * @param  _allocationPrecision             Precision of allocation percentage
+     * @return ISetToken                        The address of the proposed nextSet
+     */
+    function determineNewAllocation(
+        uint256 _targetBaseAssetAllocation,
+        uint256 _allocationPrecision
+    )
+        external
+        returns (ISetToken)
+    {
+        // Determine nextSet units and components
+        (
+            uint256[] memory nextSetUnits,
+            address[] memory nextSetComponents
+        )= calculateNextSetParameters(
+            _targetBaseAssetAllocation,
+            _allocationPrecision
+        );
+
+        // Create new collateral set with passed components, units, and collateralNaturalUnit
+        address nextSetAddress = core.createSet(
+            setTokenFactory,
+            nextSetComponents,
+            nextSetUnits,
+            collateralNaturalUnit,
+            "",
```

Comment:
When debugging / searching on etherscan, it's actually nice for the collateral set to have a name/symbol. 

At the very least, maybe this could be passed in thru the constructor

Reply:
Guess that was a part I left unfilled. What names would you propose?